federation-js/src/composition/validate/sdl/__tests__/uniqueFieldDefinitionNames.test.ts

```diff
@@ -264,5 +264,60 @@ describe('UniqueFieldDefinitionNames', () => {
       ]);
       expect(errors).toHaveLength(0);
     });
+
+    it('object type definitions field and arguments with different type', () => {
+      const typeDefs = gql`
+        type Campaign implements Node {
+          identifier: String
+          event(identifier: String!): Event
+        }
+      `;
+
+      const [definitions] = createDocumentsForServices([
+        {
+          typeDefs,
+          name: 'serviceA',
+        },
+        {
+          typeDefs,
+          name: 'serviceB',
+        },
+      ]);
+
+      const errors = validateSDL(definitions, schema, [
+        UniqueFieldDefinitionNames,
+      ]);
+      expect(errors).toHaveLength(0);
+    });
+  });
+
+  it('object type definitions field different order', () => {
```

Comment:
Is this test just asserting that order is irrelevant for fields when determining value types? Just want to make sure I understand this test isn't related to your changes (but let's keep it!)

Reply:
yup! it was failing before we did the change to fieldDiff, oddly :)

I've updated the test to have a better name (and made the schema nicer too) :)